rfcs/packaging/communicating-convergence-status.md

```diff
@@ -0,0 +1,141 @@
+
+See open questions section.
+
+#### Decision to remove `react-internal`
+
+`@fluentui/react-internal` was created as a workound for circular dependency issues that arose when breaking out components from the `@fluentui/react` version 8 suite package into individual packages (`react-checkbox` etc).
+
+It broke the cycles but left a very convoluted developer experience for v8 components: the majority lived in `react-internal`, a handful lived in individual component packages like `react-checkbox`, and a handful which used components from the individual packages had to live in `react`. From the perspective of anyone but ~4 team members who worked on the split, there was no rhyme or reason for what lived where, and building/testing became more painful.
+
+However, more recent developments have invalidated most of `react-internal`'s reasons for existence:
+
+- We decided individual component packages like `react-checkbox` should _only_ contain converged components, and moved the v8 versions back to the suite (eliminating most current cycles)
+  - Exception: `react-date-time` (not at all converged) is exported by the suite as of version 8, creating a cycle, but we can break this cycle as described above
+- Shift in focus to aggressively converging components, with a hard policy that they _must not_ depend on old components from either v8 or v0 (eliminating most risk of future cycles)
+
+While getting rid of `react-internal` is extra work, it seems worthwhile to get rid of the technical debt and improve the dev experience.
+
+### Open question: should `@fluentui/react-*` mean converged? (remove `react-` prefix from old packages?)
+
+See discussion later.
+
+### Add status badges to all component package readmes
+
+The badges below are generated using https://shields.io/ and link to a placeholder wiki page which will later contain explanations of each status.
+
+We definitely want badges for stability, possibly with sub-badges for convergence status:
+
+- [![in development](https://img.shields.io/badge/stability-in%20development-orange)](https://github.com/microsoft/fluentui/wiki/Package-status)
+- [![preview](https://img.shields.io/badge/stability-preview-yellow)](https://github.com/microsoft/fluentui/wiki/Package-status)
+  - Possible sub-badge: [![preview (legacy)](<https://img.shields.io/badge/stability-preview%20(legacy)-yellow>)](https://github.com/microsoft/fluentui/wiki/Package-status)
+- [![stable](https://img.shields.io/badge/stability-stable-green)](https://github.com/microsoft/fluentui/wiki/Package-status)
+  - Possible sub-badge: [![stable (legacy)](<https://img.shields.io/badge/stability-stable%20(legacy)-green>)](https://github.com/microsoft/fluentui/wiki/Package-status)
+- [![deprecated](https://img.shields.io/badge/stability-deprecated-red)](https://github.com/microsoft/fluentui/wiki/Package-status)
+
+We might also want separate badges for category/convergence/??? (naming suggestions welcome)
```

Comment:
I prefer the separate badges for convergence (if we need them).

Reply:
suggestion: I'd add explicit annotations to public APIS as well - ideally have automation that would take that as source of truth and generate those badges.

eg:
```js
/**
 * @experimental
 */
export function Button(props){....}

/**
 * @stable
 */
export function Avatar(props){....}
```

Reply:
I was going to suggest using the [`@packageDocumentation` comment](https://api-extractor.com/pages/tsdoc/tag_packagedocumentation/) for this purpose and including a status annotation in the comment, but at least according to the API Extractor docs that's explicitly *not* supported. 🙄 

On individual APIs we could use [release tags](https://api-extractor.com/pages/tsdoc/doc_comment_syntax/#release-tags) but that has to be done on each API individually. While this is definitely something we should consider using, I'm not sure offhand how you'd set up automation to determine the overall release status of a particular package based on individual API tags.

Another possibility for communicating release status is to just use semver, like staying on a prerelease (`-dev` followed by `-beta`?) until it's ready. I guess in theory `0.x` is also supposed to communicate this but it's a bit less explicit.

Reply:
yeah those annotations will definitely not "solve" the general status. but it's always a good idea to include them for particular apis to explicitly communicate.

I'm all for semver and prerelase tags 👍

Reply:
We may be too quick to "release" unstable/semi-stable things to partners--okay during iteration but we should keep the heaviest churn internal.

`@alpha`/`@beta` tags on individual APIs could be good if we have unstable APIs within an overall stable package, but having to add annotations on every API in a new package is less useful.

Reply:
Should ensure that before we "ship" anything, need agreement from relevant parties on both sub-teams.

Consider quarterly planning for breaks and new releases for converged stuff only (separate RFC?). Won't affect v0 or v8.

Reply:
Make an RFC or something discussing previous issues we hit with versioning a suite package.